add session docs (#34)
dimagi/open-chat-studio-docs: docs/concepts/experiment/sessions.md

## History Management

- As conversations progress, all previous messages within a session are stored as `history`.
- If the session history exceeds a predefined maximum length, it is **summarized**, and the bot will only receive:
  - A summary of older interactions.
  - The most recent exchanges to maintain context.

## Participant Data

Outside of session-specific data, Open Chat Studio maintains **participant data**, which:

- Persists across sessions.
- Is tied to the same `User, Channel, Chatbot` scope.
- Helps retain long-term user preferences and contextual information beyond a single session.

## Anonymous Sessions

On the **Web** channel, users can have **anonymous sessions**, where:

- Participant data is only available for the duration of the session.
- Since user identity cannot be verified, data cannot persist beyond the session.

## Resetting Sessions

For **Single-Session Channels** like **WhatsApp** and **Telegram**, the current session continues indefinitely unless the user sends the `/reset` command. This command:

- Ends the current session.
- Starts a new session.

> This means that, except for participant data, the bot 'forgets' all details of the previous conversation—including the fact that it even took place.

Author: ```suggestion
This means that, aside from participant data, the bot loses all information about the previous conversation — including the fact that it even took place.
```
